src/dal/views.py

```diff
@@ -25,9 +26,16 @@ class ViewMixin(object):
 
     def dispatch(self, request, *args, **kwargs):
         """Set :py:attr:`forwarded` and :py:attr:`q`."""
-        self.forwarded = json.loads(request.GET.get('forward', '{}'))
-        if request.method == 'POST':
-            self.forwarded = json.loads(request.POST.get('forward', '{}'))
+        try:
+            self.forwarded = json.loads(request.GET.get('forward', '{}'))
+            if request.method == 'POST':
+                self.forwarded = json.loads(request.POST.get('forward', '{}'))
+
+            if not isinstance(self.forwarded, dict):
+                raise ValueError
+        except ValueError:
+            return HttpResponseBadRequest('Invalid JSON data')
```

Comment:
I was playing with a fix according to your comment but then I figured that only GET and POST are supported here so maybe we want to return a `405 Method Not Allowed` if the method is not supported and that would mean implementing `OPTION` as well and then this pretty much resemble the Django default `View`. Should I modify this view to subclass it directly ?